### PR TITLE
Adding gateway-api.sigs.k8s.io for Gateway API

### PR DESCRIPTION
**Pull request by robscott**
Closely related to https://github.com/kubernetes/org/issues/2506

### CODE REVIEW

File: dns/zone-configs/k8s.io._0_base.yaml [hunk 1]
```diff
@@ -313,3 +313,8 @@ kustomize.sigs:
 kubectl.sigs:
   type: CNAME
   value: cli-experimental.netlify.app.
+# https://github.com/kubernetes-sigs/gateway-api docs (@bowei @hbagdi @danehans @jpeach @robscott)
+kustomize.sigs:
```
Comment (hbagdi) [inline dns/zone-configs/k8s.io._0_base.yaml:317]:
```suggestion
gateway-api.sigs:
```

Comment (robscott) [reply dns/zone-configs/k8s.io._0_base.yaml]: 🤦 woops, thanks for catching that!

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**